src/core/net/netif.hpp

```diff
@@ -358,6 +358,8 @@ class Netif : public InstanceLocator, public LinkedListEntry<Netif>, private Non
      * This method is intended for addresses internal to OpenThread. The @p aAddress instance is directly added in the
      * unicast address linked list.
      *
+     * If @p aAddress is already added, the call to `AddUnicastAddress()` with the same address will perform no action.
```

Comment:
Add a similar comment for RemoveUnicastAddress when the address is already removed?